python_ta/transforms/type_inference_visitor.py

```diff
@@ -215,6 +209,17 @@ def visit_boolop(self, node):
         else:
             node.type_constraints = TypeInfo(Any)
 
+    # TODO: Add check in the set_compare_type_constraints as in BinOp.
+    def visit_compare(self, node):
+        """Compare operators includes:
+        '<', '>', '==', '>=', '<=', '!='. """
+        function_type = self.type_store.lookup_function(op_to_dunder_binary(node.ops[0][0]),
```

Comment:
You aren't interpreting `node.ops` correctly. It's for chained comparisons. Please consult https://docs.python.org/3/reference/expressions.html#comparisons.

Also it would be good to improve the documentation in the `nodes/Compare.py` file.